### PR TITLE
Feature/markdown cmake gen info

### LINKED ISSUE #7525: [question] CLI command to discover the cmake_find_package filename from the Conan package name?
The `cmake_find_package` generator writes a Find Module file that defines a target in a namespace. Each of these three names (filename, namespace, target) are [configurable](https://github.com/conan-io/conan/issues/7254) in `cpp_info`. For example,

```python
        self.cpp_info.filename['cmake_find_package'] = 'BoostFile'
        self.cpp_info.names['cmake_find_package'] = 'BoostNamespace'
        self.cpp_info.name = 'BoostTarget'
```

will generate a file named `FindBoostFile.cmake` that is used like this:

```cmake
find_package(BoostFile)
target_link_libraries(${target} BoostNamespace::BoostTarget)
```

Is there a CLI command that can print out these values given a package reference? Something like

```
$ conan inspect -a cpp_info.names.cmake_find_package boost/1.73.0@
```

### PR DESCRIPTION
**Pull request by uilianries**
Changelog: Feature: List `cpp_info.names` and `cpp_info.filenames` in JSON and Markdown generator.
Docs: https://github.com/conan-io/docs/pull/1844

Closes #7525

Now anyone can read easily what will be produced when mixing cpp_info.name + cpp_info.names + cpp_info.filenames. The current feature works good and solves the cmake targets, but is almost unreadable when you don't know the feature in details. I hope these generators can show the final result without much knowledgement about the feature.

- [ ] Refer to the issue that supports this Pull Request.
- [ ] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [ ] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [ ] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Review by memsharded — APPROVED** [review]
Looking good, thanks

File: conans/client/generators/markdown.py [hunk 1]
```diff
@@ -53,17 +53,21 @@
 
 generator_cmake_find_package_tpl = textwrap.dedent("""
     ### Generator ``cmake_find_package``
-    {% set cmake_find_package_filename = cpp_info.get_filename("cmake_find_package") %}
-    {% set cmake_find_package_name = cpp_info.get_name("cmake_find_package") %}
-
-    Generates the file Find{{ cmake_find_package_filename }}.cmake
+    Generates the file Find{{ cpp_info.get_filename("cmake_find_package") }}.cmake
```
Comment (memsharded) [inline conans/client/generators/markdown.py]: Why not use those jinja variables, and be calling ``cpp_info.xxxx`` several times?

Comment (uilianries) [reply conans/client/generators/markdown.py]: Yes, it makes sense now after the conflict, I gonna update it.

Comment (uilianries) [reply conans/client/generators/markdown.py]: Done

**Review by memsharded — APPROVED** [review]

**Pull request merged**